learn/models/conformal-regression/index.qmd

```diff
@@ -0,0 +1,441 @@
+
+However, we also know that resampling methods can serve the same purpose. During resampling, we can compute a set of predictions that were not used to fit the model. For example, in 10-fold cross-validation, there are 10 collections of held-out predictions that do not suffer from the severe bias that occurs when we simple re-predict the training set. 
+
+The CV+ estimator ([Barber _et al._ (2021)](https://scholar.google.com/scholar?hl=en&as_sdt=0%2C5&q=%22Predictive+inference+with+the+jackknife%2B%22&btnG=)) can be used to assemble a reference distribution of residuals. If we use `fit_resamples()` or one of the `tune_*()` functions, we can gather those residuals and use them to create prediction intervals. We'll have to make sure that we save the out-of-sample predictions as well as the resasmpled models.  
+
+The `control_*()` functions can be used for this purpose. We can set `save_pred = TRUE` to save the predicted values. For the resampled models, we can use the tools to [extract the models](https://tidymodels.github.io/tidymodels_dot_org/learn/index.html#category=extracting%20results) via the `extract` argument. We'll use the `I()` function to return the fitted workflows from each resample: 
+
+```{r}
+#| label: control
+ctrl <- control_resamples(save_pred = TRUE, extract = I)
+```
+
+Let's use 10-fold cross-validation to resample the neural network: 
+
+```{r}
+#| label: cv
+
+set.seed(493)
+folds <- vfold_cv(train_data)
+
+nnet_rs <- 
+  nnet_wflow %>% 
+  fit_resamples(folds, control = ctrl)
+
+collect_metrics(nnet_rs)
+```
+
+The model has an estimated root mean squared error of `r collect_metrics(nnet_rs) %>% dplyr::filter(.metric == "rmse") %>% pluck("mean") %>% signif(3)` and an estimated R<sup>2</sup> of `r collect_metrics(nnet_rs) %>% dplyr::filter(.metric == "rsq") %>% pluck("mean") %>% signif(3)`.
+
+We can create another object for computing the intervals:
+
+```{r}
+#| label: int-cv
+
+cv_int <- int_conformal_cv(nnet_rs)
+cv_int
+
+# Produce 90% prediction intervals
+test_cv_res <- 
+  predict(cv_int, test_data, level = 0.90) %>% 
+  bind_cols(test_data)
+
+test_cv_res %>% slice(1:5)
+```
+
+The results for this method are:
+
+```{r}
+#| label: cv-pred
+#| out-width: 70%
+#| echo: false
+#| fig-width: 6
+#| fig-height: 4
+
+test_cv_res %>% 
+  ggplot(aes(x)) +
+  geom_point(aes(y = y), alpha = 1 / 20) +
+  geom_line(
+    data = nnet_pred, aes(y = .pred),
+    linewidth = 1, col = "blue"
+  ) +
+  geom_ribbon(aes(ymin = .pred_lower, ymax = .pred_upper), 
+              col = "#D95F02", linewidth = 3 / 4, fill = NA)
+```
+
+At each point, the interval length is `r round(mean(test_cv_res$.pred_upper - test_cv_res$.pred_lower), 2)`; the previous split conformal interval width was `r round(mean(test_split_res$.pred_upper - test_split_res$.pred_lower), 2)`. This is due to the training set being larger than the calibration set. 
+
+Note that the coverage is a little better too: 
```

Comment:
Could you clarify what you mean by "better" here. I read it as "closer to 90%", is that correct?